internal/pkg/api/cloudstack.go

```diff
@@ -4,105 +4,63 @@ import (
 	"fmt"
 	"os"
 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/yaml"
 
-	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
 	"github.com/aws/eks-anywhere/pkg/templater"
 )
 
 type CloudStackConfig struct {
-	datacenterConfig    *v1alpha1.CloudStackDatacenterConfig
-	cpMachineConfig     *v1alpha1.CloudStackMachineConfig
-	workerMachineConfig *v1alpha1.CloudStackMachineConfig
-	etcdMachineConfig   *v1alpha1.CloudStackMachineConfig
+	datacenterConfig *anywherev1.CloudStackDatacenterConfig
+	machineConfigs   map[string]*anywherev1.CloudStackMachineConfig
 }
 
 type CloudStackFiller func(config CloudStackConfig)
 
 func AutoFillCloudStackProvider(filename string, fillers ...CloudStackFiller) ([]byte, error) {
-	var etcdMachineConfig *v1alpha1.CloudStackMachineConfig
-	// only to get name of control plane and worker node machine configs
-	clusterConfig, err := v1alpha1.GetAndValidateClusterConfig(filename)
-	if err != nil {
-		return nil, fmt.Errorf("the cluster config file provided is invalid: %v", err)
-	}
-
-	cpName := clusterConfig.Spec.ControlPlaneConfiguration.MachineGroupRef.Name
-	workerName := clusterConfig.Spec.WorkerNodeGroupConfigurations[0].MachineGroupRef.Name
-	cloudstackDatacenterConfig, err := v1alpha1.GetCloudStackDatacenterConfig(filename)
+	cloudstackDatacenterConfig, err := anywherev1.GetCloudStackDatacenterConfig(filename)
```

Comment:
u can simply use `cluster.ParseConfigFromFile(filename)` to get the config obj that has everything

https://github.com/aws/eks-anywhere/blob/main/internal/pkg/api/snow.go#L22

Reply:
works for me! Thanks